swift-sdk/Internal/InternalIterableAPI.swift

```diff
@@ -73,7 +73,9 @@ final class InternalIterableAPI: NSObject, PushTrackerProtocol, AuthProvider {
     }()
     
     lazy var embeddedMessagingManager: IterableEmbeddedMessagingManagerProtocol = {
-        self.dependencyContainer.createEmbeddedMessagingManager()
+        self.dependencyContainer.createEmbeddedMessagingManager(config: self.config,
+                                                                apiClient: self.apiClient,
+                                                                dateProvider: self.dateProvider)
```

Comment:
Curious why the constructor needs date provider? Is embdeddedManager going to use that inside for timecheck?

Reply:
Yep, we also pass it in for testing (for mocking time/date).